soh/src/code/z_eff_blure.c

```diff
@@ -199,100 +200,67 @@ void EffectBlure_Init2(void* thisx, void* initParamsx) {
 void EffectBlure_Destroy(void* thisx) {
 }
 
+void updateTrailColor(const char* trailName, Color_RGBA8* color, int* changed, int* reset) {
+    Color_RGBA8 defaultColor = { 255, 255, 255, 255 };
+    char cvarName[50];
+    if (CVarGetInteger(snprintf(cvarName, sizeof(cvarName), "%s.Changed", trailName), 0)) {
+        *color = CVarGetColor(snprintf(cvarName, sizeof(cvarName), "%s.Value", trailName), defaultColor);
+        *changed = 1;
+    } else if (*changed) {
+        *color = defaultColor;
+        *reset = 1;
+    }
+}
+
+void updateParticleColors(Color_RGBA8* color, Color_RGBA8 p1StartColor, Color_RGBA8 p2StartColor,
+                          Color_RGBA8 p1EndColor, Color_RGBA8 p2EndColor) {
+    p1StartColor.r = color->r;
+    p2StartColor.r = color->r * 0.8f;
+    p1EndColor.r = color->r * 0.6f;
+    p2EndColor.r = color->r * 0.4f;
+
+    p1StartColor.g = color->g;
+    p2StartColor.g = color->g * 0.8f;
+    p1EndColor.g = color->g * 0.6f;
+    p2EndColor.g = color->g * 0.4f;
+
+    p1StartColor.b = color->b;
+    p2StartColor.b = color->b * 0.8f;
+    p1EndColor.b = color->b * 0.6f;
+    p2EndColor.b = color->b * 0.4f;
+}
+
 s32 EffectBlure_Update(void* thisx) {
     EffectBlure* this = (EffectBlure*)thisx;
     s32 i;
     Color_RGBA8 color;
     static u8 changed = 0;
     u8 reset = 0;
 
-    switch (this->trailType) { //there HAS to be a better way to do this.
```

Comment:
It looks like this PR is incorporating changes from another PR that are unrelated to this one.  Specifically, these changes here appear to be from #3973.  Would it be possible to rebase this PR or open a new one with only the changes you intended to make for this PR?

Reply:
Oh yikes, sorry about that. I'll do that right now